src/library/ButtonGroup/ButtonGroup.js

```diff
@@ -45,17 +45,42 @@ type State = {
 export const componentTheme = (baseTheme: Object) => ({
   ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
   ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
+  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
+  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
   ButtonGroupButton_borderStartColor: baseTheme.borderColor,
   ButtonGroupButton_borderStartColor_checked: 'currentcolor',
   ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,
 
   ...baseTheme
 });
 
-const styles = ({ fullWidth, theme: baseTheme }) => {
+const styles = ({ fullWidth, theme: baseTheme, variant }) => {
   let theme = componentTheme(baseTheme);
+  if (variant) {
+    // prettier-ignore
+    theme = {
+      ...theme,
+      ButtonGroupButton_borderColor_active: theme[`borderColor_${variant}_active`],
+      ButtonGroupButton_borderColor_hover: theme[`borderColor_${variant}_hover`]
+    };
+  }
   const { direction } = theme;
   const rtl = direction === 'rtl';
+  const borderStartColorProperty = rtl ? 'borderRightColor' : 'borderLeftColor';
```

Comment:
Consider the following update to simplify and gain alignment in variable names with respective css properties.

```js
  const start = rtl ? 'Right' : 'Left';
  const end = rtl ? 'Left' : 'Right';
  
  const borderStartColorProperty = `border${start}Color`;
  const borderEndColorProperty = `border${end}Color`;
    
  const borderTopStartRadiusProperty = `borderTop${start}Radius`;
  const borderTopEndRadiusProperty = `borderTop${end}Radius`;
  
  const borderBottomStartRadiusProperty = `borderBottom${start}Radius`;
  const borderBottomEndRadiusProperty = `borderBottom${end}Radius`;
```

Reply:
Ooh... ignore my comment, this is much better.

Reply:
Yes! I had started thinking about something like this, but I was deep in the weeds and struggled to come up with. I really like this approach, thanks!